SpatialConvolution.lua

```diff
@@ -90,243 +87,68 @@ function SpatialConvolution:noBias()
    return self
 end
 
-function SpatialConvolution:createIODescriptors(input)
-    local batch = true
-    if input:dim() == 3 then
-        input = input:view(1, input:size(1), input:size(2), input:size(3))
-        batch = false
-    end
-    assert(input:dim() == 4 and input:isContiguous());
-    self.iSize = self.iSize or torch.LongStorage(4):fill(0)
-    if not self.iDesc or not self.oDesc or
-        input:size(1) ~= self.iSize[1] or input:size(2) ~= self.iSize[2]
-    or input:size(3) ~= self.iSize[3] or input:size(4) ~= self.iSize[4] then
-        self.iSize = input:size()
 
-        assert(self.nInputPlane == input:size(2), 'input has to contain: '
-                   .. self.nInputPlane
-                   .. ' feature maps, but received input of size: '
-                   .. input:size(1) .. ' x ' .. input:size(2) ..
-                   ' x ' .. input:size(3) .. ' x ' .. input:size(4))
+function SpatialConvolution:checkInputChanged(input)
+    assert(input:isContiguous())
+
+    if not self.iSize or self.iSize:size() ~= input:dim() then
+       self.iSize = torch.LongStorage(input:dim()):fill(0)
+    end
+    self.groups = self.groups or 1
+    if not self.weightDesc then self:resetWeightDescriptors() end
+    if not self.iDesc or not self.oDesc or input:size(1) ~= self.iSize[1] or input:size(2) ~= self.iSize[2]
+    or input:size(3) ~= self.iSize[3] or input:size(4) ~= self.iSize[4] or (input:dim()==5 and input:size(5) ~= self.iSize[5]) then
```

Comment:
you added a (input:dim()==5 and input:size(5) ~= self.iSize[5])
but you did not chnage the error message in the assertion below. fix that error message to incorporate teh 5D case as well.